### PR TITLE
[REVIEWING] Add kyberswap elastic pool v2

### PR DESCRIPTION
**Pull request by nhd98z**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by nhd98z** [comment]
@Hosuke @jeff-dude please review this pr, quite straight forward.

File: models/kyberswap/avalanche_c/kyberswap_avalanche_c_schema.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ models:
       - check_dex_seed:
           blockchain: avalanche_c
           project: kyberswap
-          version: dmm
+          version: elastic_2
```
Comment (Hosuke) [inline models/kyberswap/avalanche_c/kyberswap_avalanche_c_schema.yml:26]: Do you have the corresponding seed added in this file:
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/kyberswap/avalanche_c/kyberswap_avalanche_c_schema.yml (hunk 1 above)
Comment (Hosuke) [reply models/kyberswap/avalanche_c/kyberswap_avalanche_c_schema.yml:26]: I suggest we keep the `dmm`, and also add one new `check_dex_seed` test for `elastic_2`.

Comment (nhd98z) [reply models/kyberswap/avalanche_c/kyberswap_avalanche_c_schema.yml:26]: @Hosuke `dmm` should be called `classic`, I've changed it to `version: classic`, pls check.

File: models/kyberswap/arbitrum/kyberswap_arbitrum_trades.sql [hunk 2]
```diff
@@ -65,8 +65,31 @@ kyberswap_dex AS (
...
+        ,cast(if(starts_with(cast(t.deltaQty0 as varchar), '-'), abs(t.deltaQty1), abs(t.deltaQty0)) as uint256)  AS token_bought_amount_raw
+        ,cast(if(starts_with(cast(t.deltaQty0 as varchar), '-'), abs(t.deltaQty0), abs(t.deltaQty1)) as uint256)  AS token_sold_amount_raw
+        ,NULL                                                                                                     AS amount_usd
+        ,if(starts_with(cast(t.deltaQty0 as varchar), '-'), p.token1, p.token0)                                   AS token_bought_address
+        ,if(starts_with(cast(t.deltaQty0 as varchar), '-'), p.token0, p.token1)                                   AS token_sold_address
+        ,t.contract_address                                                                                       AS project_contract_address
+        ,t.evt_tx_hash                                                                                            AS tx_hash
+        ,'elastic_2'                                                                                              AS version
```
Comment (Hosuke) [inline models/kyberswap/arbitrum/kyberswap_arbitrum_trades.sql:82]: Do you want to add some seed for trades with `elastic_2` version?

File: models/kyberswap/avalanche_c/kyberswap_avalanche_c_trades.sql [hunk 3]
```diff
@@ -53,6 +54,7 @@ kyberswap_dex AS (
         ,if(starts_with(cast(t.deltaQty0 as varchar), '-'), p.token0, p.token1)                                  AS token_sold_address
         ,t.contract_address                                                                                      AS project_contract_address
         ,t.evt_tx_hash                                                                                           AS tx_hash
+        ,'elastic'                                                                                               AS version
```
Comment (Hosuke) [inline models/kyberswap/avalanche_c/kyberswap_avalanche_c_trades.sql:57]: And for `elastic` version here?

File: models/kyberswap/polygon/kyberswap_polygon_schema.yml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index
+            - trace_address
+      - check_dex_aggregator_seed:
+          blockchain: polygon
+          project: kyberswap
+          version: meta_2
```
Comment (Hosuke) [inline models/kyberswap/polygon/kyberswap_polygon_schema.yml:117]: Also for `meta_2` version.
Without the seed, the `check_dex_aggregator_seed` will not work.

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/kyberswap/arbitrum/kyberswap_arbitrum_trades.sql (hunk 2 above)
Comment (nhd98z) [reply models/kyberswap/arbitrum/kyberswap_arbitrum_trades.sql:82]: No we can skip it, I just update the schema to pass the test.

File: models/kyberswap/avalanche_c/kyberswap_avalanche_c_trades.sql (hunk 3 above)
Comment (nhd98z) [reply models/kyberswap/avalanche_c/kyberswap_avalanche_c_trades.sql:57]: No we can skip it, I just update the schema to pass the test.

File: models/kyberswap/polygon/kyberswap_polygon_schema.yml (hunk 4 above)
Comment (nhd98z) [reply models/kyberswap/polygon/kyberswap_polygon_schema.yml:113]: I did have in `https://github.com/duneanalytics/spellbook/blob/7e9111697e045aad0e7cf32b2fad5a8f172c412c/seeds/dex/aggregator/dex_aggregator_seed.csv#L116`

**Comment by nhd98z** [comment]
@Hosuke Hmm why I update `dmm` to `classic` in both dex seed data and sql but still fail test?

**Review by Hosuke — CHANGES_REQUESTED** [review]
These are unmatched seed records:
https://dune.com/queries/3095266

One possible cause may be these records are not in the query reults.
Please verify.

**Comment by nhd98z** [comment]
@Hosuke I think the test case is out of date, I read the transaction from 1 years ago from weird contract and don't get it. Better remove the old test (classic + elastic). Our elastic_2 is hold the main liquidity.

**Comment by Hosuke** [comment]
> @Hosuke I think the test case is out of date, I read the transaction from 1 years ago from weird contract and don't get it. Better remove the old test (classic + elastic). Our elastic_2 is hold the main liquidity.

In this case, do you want to add some more new seeds for `elastic_2`?

**Comment by nhd98z** [comment]
> > @Hosuke I think the test case is out of date, I read the transaction from 1 years ago from weird contract and don't get it. Better remove the old test (classic + elastic). Our elastic_2 is hold the main liquidity.
> 
> In this case, do you want to add some more new seeds for `elastic_2`?

Yep just add one, things look good. Because `elastic_2` has same interface as `elastic`, we just fix the bug with openzeppelin.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @nhd98z

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**